src/mds/ScrubStack.cc

```diff
@@ -98,9 +98,51 @@ int ScrubStack::_enqueue(MDSCacheObject *obj, ScrubHeaderRef& header, bool top)
   return 0;
 }
 
+void ScrubStack::purge_scrub_counters(std::string_view tag)
+{
+  for (auto& stat : mds_scrub_stats) {
+    if (tag == "all") {
+      stat.counters.clear();
+    } else {
+      auto it = stat.counters.find(std::string(tag));
+      if (it != stat.counters.end()) {
+	stat.counters.erase(it);
+      }
+    }
+  }
+}
+
+// called from tick
+void ScrubStack::purge_old_scrub_counters()
+{
+  // "mds_scrub_stats_review_period" must be in number of days
+  uint64_t mds_scrub_stats_review_period = g_conf().get_val<uint64_t>("mds_scrub_stats_review_period");
```

Comment:
Maybe you could save this in a local member instead of reading this from the `g_conf()` per `tick()` ?

Reply:
I'd thought of that too, but then didn't want to burden myself about how to handle changes to this config item at run time.

Reply:
There were several commits improved the code like this in `mds/client`.

Reply:
is it okay if I add a new commit to handle this via a `handle_conf_change()` method ?